modules/rostests/apitests/win32nt/ntuser/NtUserSetTimer.c

```diff
@@ -1,18 +1,293 @@
 /*
- * PROJECT:         ReactOS api tests
- * LICENSE:         GPL - See COPYING in the top level directory
- * PURPOSE:         Test for NtUserSetTimer
- * PROGRAMMERS:
+ * PROJECT:     ReactOS API tests
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Tests for NtUserSetTimer
+ * COPYRIGHT:   Copyright 2008 Timo Kreuzer <timo.kreuzer@reactos.org>
+ *              Copyright 2024 Tomas Vesely <turican0@gmail.com>
  */
 
 #include "../win32nt.h"
 
+#define SLEEP_TIME 500
+#define TIME_TOLERANCE 0.5
+
+#define TEST1_COUNT 20
+#define TEST1_INTERVAL 10
+
+#define TEST2_COUNT 20
+#define TEST2_INTERVAL 10
+
+#define TEST3_COUNT 40000
+#define TEST3_INTERVAL 10
+
+#define TEST4_COUNT 40000
+#define TEST4_INTERVAL 10
+
+typedef struct TIMER_MESSAGE_STATE1
+{
+    UINT_PTR index;
+    int counter;
+} TIMER_MESSAGE_STATE1;
+
+TIMER_MESSAGE_STATE1 timerId1[TEST1_COUNT];
+
+typedef struct TIMER_MESSAGE_STATE2
+{
+    int counter;
+} TIMER_MESSAGE_STATE2;
+
+TIMER_MESSAGE_STATE2 timerId2[TEST2_COUNT];
+
+void CALLBACK TimerProc(HWND hWnd, UINT uMsg, UINT_PTR idEvent, DWORD dwTime)
+{
+    for (int i = 0; i < TEST1_COUNT; i++)
+        if (timerId1[i].index == idEvent)
+            timerId1[i].counter++;
+}
+
+void MessageLoop(void)
+{
+    MSG msg;
+    while (GetMessage(&msg, NULL, 0, 0))
+    {
+        TranslateMessage(&msg);
+        DispatchMessage(&msg);
+    }
+}
+
+LRESULT CALLBACK WindowProc(HWND hwnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
+{
+    switch (uMsg)
+    {
+    case WM_TIMER:
+        timerId2[wParam].counter++;
+        return 0;
+    }
+    return DefWindowProc(hwnd, uMsg, wParam, lParam);
+}
+
+BOOL test1(void)
+{
+    int countErrors = 0;
+
+    int minMessages = (SLEEP_TIME / TEST1_INTERVAL) * (1 - TIME_TOLERANCE);
+    int maxMessages = (SLEEP_TIME / TEST1_INTERVAL) * (1 + TIME_TOLERANCE);
+
+    for (int i = 0; i < TEST1_COUNT; i++)
```

Comment:
```suggestion
    for (i = 0; i < TEST1_COUNT; i++)
```

Reply:
I think you could just do: `ZeroMemory(timerId1, sizeof(timerId1));` here instead of this loop.